platform/pulpcore/app/viewsets/content.py

```diff
@@ -44,6 +45,28 @@ class ContentViewSet(NamedModelViewSet):
     serializer_class = ContentSerializer
     filter_class = ContentFilter
 
+    @transaction.atomic
+    def create(self, request):
+        """
+        Create a new Content instance from
+        :param request:
+        :return:
+        """
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        artifacts = serializer.validated_data.pop('artifacts')
+        content = serializer.save()
+
+        for relative_path, artifact in artifacts.items():
+            ca = ContentArtifact(artifact=artifact, content=content, relative_path=relative_path)
+            ca.save()
+
+        headers = self.get_success_headers(serializer.data)
+        return Response(serializer.data, status=status.HTTP_201_CREATED, headers=headers)
+
+    def update(self, request, pk):
```

Comment:
It would be better if we can find a way for this to not show up as an option in the API's metadata, such as when a user makes a METHODS request to this endpoint, or just DRF's knowledge of the API.

Sadly it looks like the options are limited. I think there may be two:

1. override the ``as_view()`` method and remove the "update" value from the "actions" argument before passing it through to the superclass.
2. delete the method from the class. This isn't great, but that seems to be how DRF detects capabilities. So under the class definition, I think ``del ContentViewSet.update`` would work.

Maybe you can come up with a better idea. I'm a little surprised DRF doesn't give us an easy way to control this. Their guidance is to use mixings to assemble only the capabilities you want, but that doesn't work so well when we're extending a ViewSet with our own boilerplate.